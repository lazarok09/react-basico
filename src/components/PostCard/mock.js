export const postCardMockProps = {
  title: "title example",
  body: "paragrafo",
  id: 1,
  photo: "img/img.png",
};
