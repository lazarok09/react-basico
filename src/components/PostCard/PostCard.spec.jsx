import { render, screen } from '@testing-library/react';
import { PostCard } from '.';
import { postCardMockProps } from './mock';
// cria dados falsos
const props = postCardMockProps;

describe('<PostCard />', () => {
  it('should render PostCard correctly', () => {
    render(<PostCard {...props} />);
    expect(screen.getByAltText('title example')).toHaveAttribute('src', 'img/img.png');
    expect(screen.getByRole('heading', { name: 'title example' })).toBeInTheDocument();

    expect(screen.getByText('paragrafo')).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { container } = render(<PostCard {...props} />);
    expect(container.firstChild).toMatchSnapshot();
    // primeiro filho do elemento Ã© o elemento que estou renderizando
  });
});
