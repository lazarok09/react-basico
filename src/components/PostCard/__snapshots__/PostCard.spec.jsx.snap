// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should match snapshot 1`] = `
<div
  class="post-card"
>
  <img
    alt="title example"
    src="img/img.png"
  />
  <h2>
    title example
  </h2>
  <p>
    paragrafo
  </p>
</div>
`;
